### PR TITLE
docs(common): Add Authentication with NuGet Keys

### PR DESCRIPTION
**Pull request by yordan-mitev**
This PR adds the required documentation for the newly added [NuGet Keys feature](https://www.telerik.com/blogs/announcing-nuget-keys) in Your Account.

I discussed this topic with @dimodi before creating the PR.

### CODE REVIEW

File: deployment/nuget-keys.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+---
+title: Restoring NuGet Packages in CI
+page_title: Restoring NuGet Packages in CI
+description: "Get started with Telerik UI for Blazor and use NuGet Keys to authenticate with the Telerik NuGet server and restore packages in your CI or desktop environment."
```
Comment (dimodi) [inline deployment/nuget-keys.md:4]: Do we need quotation marks here? I haven't seen them being used elsewhere.

File: deployment/nuget-keys.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+slug: installation/nuget-keys
+position: 7
+---
+
+# Restoring NuGet Packages in Your CI Workflow
+
+This article provides an overview of the most popular approaches for using token-based authentication to restore Telerik NuGet packages in your CI (Continuous Integration) workflow.
```
Comment (dimodi) [inline deployment/nuget-keys.md:11]:
```suggestion
This article provides an overview of the most popular ways to use token-based authentication to restore Telerik NuGet packages in your Continuous Integration (CI) workflow.
```

File: deployment/nuget-keys.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+Unlike your Telerik credentials, a NuGet Key has a limited scope and can be used only with the Telerik NuGet server. If any of your NuGet keys is compromised, you can quickly delete it and create a new one.
+
+## Generating NuGet Keys
+
+1. Go to the [**Manage NuGet Keys**](https://www.telerik.com/account/downloads/nuget-keys) page in your Telerik account.
+
+    ![Manage NuGet Keys](../deployment/images/manage-nuget-keys.png)
+
+1. To create a new key, select the **Generate New Key** button.
```
Comment (dimodi) [inline deployment/nuget-keys.md:28]:
```suggestion
1. To create a new key, click on **Generate New Key**.
```

File: deployment/nuget-keys.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+1. To copy the key, select **Copy and Close**. Once you close the window, you can no longer copy the generated key. For security reasons, the **NuGet Keys** page displays only a portion of the key.
+
+    ![Copy Generated NuGet Key](../deployment/images/copy-nuget-key.png)
+
+## Storing a NuGet Key
+
+> Never check in a NuGet Key with your source code or leave it publicly visible in plain text, for example, as a raw key value in a `nuget.config` file. A NuGet Key is valuable as bad actors can use it to access the NuGet packages that are licensed under your account. A potential key abuse could lead to a review of the affected account.
+
+To protect the NuGet Key, store it as a secret environment variable. The exact steps depend on your workflow:
+
```
Comment (dimodi) [inline deployment/nuget-keys.md:41]: Maybe we should use a table here, to highlight the different environments. Or at least bold them.

**Review by dimodi — APPROVED** [review]
I have added a few comments and suggestions for your consideration.

File: deployment/nuget-keys.md (hunk 1 above)
Comment (yordan-mitev) [reply deployment/nuget-keys.md:4]: You're right, the Blazor docs don't use quotation marks in the description.
```suggestion
description: Get started with Telerik UI for Blazor and use NuGet Keys to authenticate with the Telerik NuGet server and restore packages in your CI or desktop environment.
```

File: deployment/nuget-keys.md (hunk 2 above)
Comment (yordan-mitev) [reply deployment/nuget-keys.md:11]: For completeness, the article mentions one other way—basic authentication. But it handles in detail only the token-based authentication through NuGet keys and the workflows where these keys can be used.

```suggestion
This article provides an overview of the most popular ways for using token-based authentication to restore Telerik NuGet packages in your Continuous Integration (CI) workflow.
```

File: deployment/nuget-keys.md (hunk 4 above)
Comment (yordan-mitev) [reply deployment/nuget-keys.md:41]: I am hesitant about using a table with three rows. There is no need to bold them for emphasis as these workflows are at the beginning of the list items.

**Pull request merged**